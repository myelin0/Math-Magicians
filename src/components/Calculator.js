import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const { total, next, operation } = state;

  const clickHandler = (e) => {
    const data = calculate(state, e.target.textContent);
    setState(data);
  };
  return (
    <>
      <div className="calcWrapper">
        <h2>Let`s do some math!</h2>
        <div className="container">
          <div className="calc-grid">
            <div className="output">
              {`${total || ''} ${operation || ''} ${next || ''}`}
            </div>
            <div
              className="input"
              aria-hidden
              onClick={clickHandler}
            >
              <button type="button">AC</button>
              <button type="button">+/-</button>
              <button type="button">%</button>
              <button type="button" className="orange">รท</button>
              <button type="button">7</button>
              <button type="button">8</button>
              <button type="button">9</button>
              <button type="button" className="orange">x</button>
              <button type="button">4</button>
              <button type="button">5</button>
              <button type="button">6</button>
              <button type="button" className="orange">-</button>
              <button type="button">1</button>
              <button type="button">2</button>
              <button type="button">3</button>
              <button type="button" className="orange">+</button>
              <button type="button" className="span-two">0</button>
              <button type="button">.</button>
              <button type="button" className="orange">=</button>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default Calculator;
