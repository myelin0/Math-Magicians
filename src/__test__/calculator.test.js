import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Calculator from '../components/calculator';
import '@testing-library/jest-dom';

it('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Perform calculation operations', () => {
  test('Click a number and see on screen', () => {
    render(<Calculator />);
    const result = screen.getByTestId('screen-output');
    userEvent.click(screen.getByText('1'));
    expect(result.innerHTML).toMatch('1');
  });
  test('Click two numbers and see on screen', () => {
    render(<Calculator />);
    const result = screen.getByTestId('screen-output');
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('3'));
    expect(result.innerHTML).toMatch('23');
  });
  test('Click equals button to see result of operation on screen', () => {
    render(<Calculator />);
    const result = screen.getByTestId('screen-output');
    userEvent.click(screen.getByText('8'));
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('+'));
    userEvent.click(screen.getByText('3'));
    userEvent.click(screen.getByText('='));
    expect(result.innerHTML).toMatch('85');
  });
  test('Click modulus button to see remainder of two numbers on screen', () => {
    render(<Calculator />);
    const result = screen.getByTestId('screen-output');
    userEvent.click(screen.getByText('AC'));
    userEvent.click(screen.getByText('4'));
    userEvent.click(screen.getByText('0'));
    userEvent.click(screen.getByText('%'));
    userEvent.click(screen.getByText('2'));
    expect(result.innerHTML).toMatch('0');
  });
});
