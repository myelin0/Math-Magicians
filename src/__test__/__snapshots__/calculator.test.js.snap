// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calcWrapper"
>
  <h2>
    Let\`s do some math!
  </h2>
  <div
    className="container"
  >
    <div
      className="calc-grid"
    >
      <div
        className="output"
        data-testid="screen-output"
      >
          
      </div>
      <div
        aria-hidden={true}
        className="input"
        onClick={[Function]}
      >
        <button
          type="button"
        >
          AC
        </button>
        <button
          type="button"
        >
          +/-
        </button>
        <button
          type="button"
        >
          %
        </button>
        <button
          className="orange"
          type="button"
        >
          รท
        </button>
        <button
          type="button"
        >
          7
        </button>
        <button
          type="button"
        >
          8
        </button>
        <button
          type="button"
        >
          9
        </button>
        <button
          className="orange"
          type="button"
        >
          x
        </button>
        <button
          type="button"
        >
          4
        </button>
        <button
          type="button"
        >
          5
        </button>
        <button
          type="button"
        >
          6
        </button>
        <button
          className="orange"
          type="button"
        >
          -
        </button>
        <button
          type="button"
        >
          1
        </button>
        <button
          type="button"
        >
          2
        </button>
        <button
          type="button"
        >
          3
        </button>
        <button
          className="orange"
          type="button"
        >
          +
        </button>
        <button
          className="span-two"
          type="button"
        >
          0
        </button>
        <button
          type="button"
        >
          .
        </button>
        <button
          className="orange"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
